api/types/plugin.go

```diff
@@ -36,6 +36,10 @@ const (
 	PluginTypeOpenAI = "openai"
 	// PluginTypeOkta is the Okta plugin
 	PluginTypeOkta = "okta"
+	// PluginTypeMDM represents MDM plugin type collectively
+	PluginTypeMDM = "mdm"
```

Comment:
This looks unused

Reply:
This will be used in the Web API to differentiate MDM plugins vs Access plugins.

Reply:
Plugin can only have one type, so would Jamf have type `mdm` or `jamf`?

What are the expected uses of grouping plugins as `mdm`? We can probably tell if the plugin is MDM "statically", e.g.

```go
func (p *PluginV1) IsMDM() bool {
    return p.GetType() == PluginTypeJamf || p.GetType() == PluginTypeSomeOtherMDM
}
```

Reply:
We can also have a label for MDM plugins, if that helps us query/filter them somehow.

Reply:
sorry I missed the input box here and replied below in comment thread [below](https://github.com/gravitational/teleport/pull/26374#issuecomment-1554533575). 

To add more, I think it will be way easy and less compute hungry to just compare string rather than the example above. I think metadata label should work as well but should those label's be used for customer specific labeling instead of internal system level labeling? I don't have strong opinion on that but just think  `PluginTypeMDM` will be simple option to group mdm types.

Reply:
`SubKind` sounds okay, I guess.

`Type` is a bit different - it is specific to Plugin, is not a field (instead derived from the `Settings` variant), and is already used for concrete types (Slack, OpenAI, etc.), rather than categories (access, MDM, etc.)

I would suggest leaving `PluginTypeJamf`, and have MDM as the SubKind. Perhaps introduce `PluginKindAccessRequest` or similar, and `PluginKindMDM`.

Reply:
Hmm, that sounds more reasonable. I'll add a new PluginSubKind types and remove MDM type from here.